From 6b44e0b376072f3964e3c33b0b0420f7f80b0472 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Mon, 24 Sep 2012 16:59:52 +0800
Subject: [PATCH 060/304] ENGR00225629 Change init.gprs-pppd's target dir to
 /etc/ppp

make_ext4fs will generated system.img based on the fs_path_config
in android_filesystem_config.h. Move the init.gprs-pppd to /etc/ppp
will make sure the init.gprs-pppd have execute permission in system.img

Change-Id: I9e2030761419f2c66df4c9afc71b8f30bc83762e
Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 imx5x/imx5x.mk              |  2 +-
 imx5x/init.rc               |  2 +-
 imx6/etc/init.gprs-pppd     | 31 -------------------------------
 imx6/etc/init.rc            |  2 +-
 imx6/etc/ppp/init.gprs-pppd | 31 +++++++++++++++++++++++++++++++
 imx6/imx6.mk                |  2 +-
 6 files changed, 35 insertions(+), 35 deletions(-)
 delete mode 100755 imx6/etc/init.gprs-pppd
 create mode 100755 imx6/etc/ppp/init.gprs-pppd

diff --git a/imx5x/imx5x.mk b/imx5x/imx5x.mk
index 37bf6ce..ea32082 100644
--- a/imx5x/imx5x.mk
+++ b/imx5x/imx5x.mk
@@ -248,7 +248,7 @@ PRODUCT_COPY_FILES +=	\
 # for all other directory
 PRODUCT_COPY_FILES +=	\
 	device/fsl/imx5x/ueventd.freescale.rc:root/ueventd.freescale.rc \
-	device/fsl/imx5x/init.gprs-pppd:system/etc/init.gprs-pppd
+	device/fsl/imx5x/init.gprs-pppd:system/etc/ppp/init.gprs-pppd
 
 # for property
 PRODUCT_DEFAULT_PROPERTY_OVERRIDES := \
diff --git a/imx5x/init.rc b/imx5x/init.rc
index 311e1a6..3294818 100644
--- a/imx5x/init.rc
+++ b/imx5x/init.rc
@@ -589,7 +589,7 @@ service ril-daemon /system/bin/rild -l /system/lib/libreference-ril.so
     user root
     group radio cache inet misc audio
 
-service pppd_gprs /etc/init.gprs-pppd
+service pppd_gprs /etc/ppp/init.gprs-pppd
     class main
     user root
     group radio cache inet misc
diff --git a/imx6/etc/init.gprs-pppd b/imx6/etc/init.gprs-pppd
deleted file mode 100755
index 7ee5e77..0000000
--- a/imx6/etc/init.gprs-pppd
+++ /dev/null
@@ -1,31 +0,0 @@
-#!/system/bin/sh
-# An unforunate wrapper script
-# so that the exit code of pppd may be retrieved
-
-
-# this is a workaround for issue #651747
-#trap "/system/bin/sleep 1;exit 0" TERM
-
-
-PPPD_PID=
-
-/system/bin/setprop "net.gprs.ppp-exit" ""
-
-/system/bin/log -t pppd "Starting pppd"
-
-#/system/xbin/pppd $*
-# pppd was put into /system/bin instead of /system/xbin after SDK1.6
-if ls /dev/ttyUSB0 > /dev/null 2>&1; then
-    /system/bin/pppd  /dev/ttyUSB0 115200 mru 1280 mtu 1280 nodetach debug dump defaultroute usepeerdns novj novjccomp noipdefault ipcp-accept-local ipcp-accept-remote connect-delay 5000 linkname ppp0
-else
-   /system/bin/pppd ttyACM0 921600 nodetach noauth noipdefault defaultroute usepeerdns linkname ppp0 connect "chat -v '' AT OK ATD*99***1# CONNECT" debug
-fi
-
-
-PPPD_EXIT=$?
-PPPD_PID=$!
-
-/system/bin/log -t pppd "pppd exited with $PPPD_EXIT"
-
-/system/bin/setprop "net.gprs.ppp-exit" "$PPPD_EXIT"
-
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 2a56286..9a61a59 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -586,7 +586,7 @@ service ril-daemon /system/bin/rild -l /system/lib/libreference-ril.so
     user root
     group radio cache inet misc audio
 
-service pppd_gprs /etc/init.gprs-pppd
+service pppd_gprs /etc/ppp/init.gprs-pppd
     class main
     user root
     group radio cache inet misc
diff --git a/imx6/etc/ppp/init.gprs-pppd b/imx6/etc/ppp/init.gprs-pppd
new file mode 100755
index 0000000..7ee5e77
--- /dev/null
+++ b/imx6/etc/ppp/init.gprs-pppd
@@ -0,0 +1,31 @@
+#!/system/bin/sh
+# An unforunate wrapper script
+# so that the exit code of pppd may be retrieved
+
+
+# this is a workaround for issue #651747
+#trap "/system/bin/sleep 1;exit 0" TERM
+
+
+PPPD_PID=
+
+/system/bin/setprop "net.gprs.ppp-exit" ""
+
+/system/bin/log -t pppd "Starting pppd"
+
+#/system/xbin/pppd $*
+# pppd was put into /system/bin instead of /system/xbin after SDK1.6
+if ls /dev/ttyUSB0 > /dev/null 2>&1; then
+    /system/bin/pppd  /dev/ttyUSB0 115200 mru 1280 mtu 1280 nodetach debug dump defaultroute usepeerdns novj novjccomp noipdefault ipcp-accept-local ipcp-accept-remote connect-delay 5000 linkname ppp0
+else
+   /system/bin/pppd ttyACM0 921600 nodetach noauth noipdefault defaultroute usepeerdns linkname ppp0 connect "chat -v '' AT OK ATD*99***1# CONNECT" debug
+fi
+
+
+PPPD_EXIT=$?
+PPPD_PID=$!
+
+/system/bin/log -t pppd "pppd exited with $PPPD_EXIT"
+
+/system/bin/setprop "net.gprs.ppp-exit" "$PPPD_EXIT"
+
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index 8ff1bfd..c238bff 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -256,7 +256,7 @@ PRODUCT_COPY_FILES +=	\
 	device/fsl/imx6/etc/apns-conf.xml:system/etc/apns-conf.xml \
 	device/fsl/imx6/etc/init.usb.rc:root/init.freescale.usb.rc \
 	device/fsl/imx6/etc/ueventd.freescale.rc:root/ueventd.freescale.rc \
-	device/fsl/imx6/etc/init.gprs-pppd:system/etc/init.gprs-pppd \
+	device/fsl/imx6/etc/ppp/init.gprs-pppd:system/etc/ppp/init.gprs-pppd \
 	device/fsl/imx6/etc/ota.conf:system/etc/ota.conf \
 	device/fsl/common/display/display_mode_fb0.conf:system/etc/display_mode_fb0.conf \
 	device/fsl/common/display/display_mode_fb2.conf:system/etc/display_mode_fb2.conf \
-- 
1.8.0

